### PR TITLE
Updated Guest.cpp per issue 12337 to remove gotos

### PR DESCRIPTION
**Pull request by batorni**
Updated Guest.cpp per issue #12337 to remove goto's from DecideAndBuyItem()

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
Replays are failing suggesting you have modified the logic of the function.

**Comment by batorni** [comment]
@duncanspumpkin I'll take a look. Is there any way to run the regression tests locally? The testing page on the wiki didn't detail anything about running those tests.

**Comment by janisozaur** [comment]
Yes, you can: just use `ctest`, after building with `WITH_TESTS=on`. The replay tests will auto-discover replays, so they need to be placed to proper data path beforehand, I think it's just `ln -s <openrct2_repo>/tests/testdata` in the dir where you have the executables built.

**Comment by duncanspumpkin** [comment]
You can also manually Replay a replay in game with the console type `replay_start <replay name>`. It's just bpb1000Ticks which is failing which is our basic busy map that has to run for 1000 ticks without the sprite data changing from the expect. If you run the replay manually then it will generate a desync report at the first and last tick telling you what fields are different.

**Comment by tupaschoal** [comment]
@batorni I'm converting this PR to draft, if you resume work and fix the failures, make sure to click "ready to review"

**Comment by Gymnasiast** [comment]
I'll have to close this PR, for the following reasons:

- The OP has not replied since 27 July
- It fails tests
- The commit history is hard to follow.

Of course, we would welcome another attempt. If you feel like doing that, please keep every `goto` elimination in its own commit, so that we can properly review it - otherwise, it's way too likely that an error slips in.
